modin/distributed/dataframe/pandas/partitions.py

```diff
@@ -54,10 +61,10 @@ def unwrap_partitions(api_layer_object, axis=None, get_ip=False):
     modin_frame._propagate_index_objs(None)
     if axis is None:
 
-        def _unwrap_partitions():
+        def _unwrap_partitions() -> list:
             [p.drain_call_queue() for p in modin_frame._partitions.flatten()]
 
-            def get_block(partition):
+            def get_block(partition: Union[PandasOnRayDataframePartition, PandasOnDaskDataframePartition]) -> np.ndarray:
```

Comment:
```suggestion
            def get_block(partition: PandasDataframePartition) -> np.ndarray:
```
maybe it's worth using the base class here for simplicity? Judging by the function's implementation it definitely can process  the `PandasDataframePartition` class itself as well as any of its descendants

Reply:
Yes we should most definitely use the base class here.

Reply:
If you look further down in this function, the function only works with these two instances. I don't see why other subclasses of `PandasDataframePartition` wouldn't work, but I wanted to keep the typing here consistent with what would really work in this function. I don't have a strong opinion either way.

Reply:
I agree with @mvashishtha.

Reply:
Actually, what's the reason for that assertion? I don't see any part in the resulting control flow that requires the partitions to be specifically either PandasOnRay or PandasOnDask variants.

Reply:
@noloerino this method has to return ray or dask futures, so it can only work with ray or dask virtual or block partitions. The virtual partitions can have multiple blocks but this method assumes there's only one block. So only the ray and dask block partitions work.

@dchigarev the assertion prevents this function from working with `PandasDataframePartition` descendant `PandasOnPythonDataframePartition`, but that aside, the docstring says it should only return ray and dask futures.

Going back to the `get_block` code in question in this PR, I still prefer keeping it as is to match the assertion. We might be able to make the list less complicated if we generalize this function to include other partition types.

Reply:
For this PR, we should keep the typing as close to actual behaviour as possible. I think we should refactor the function so that it works with any descendant of `PandasDataframePartition` (and shorten the typing) in a separate PR.